docs/temporal.md

```diff
@@ -0,0 +1,229 @@
+
+This is a simple but complete example of wrapping an agent for durable execution, creating a Temporal workflow with durable execution logic, connecting to a Temporal server, and running the workflow from non-durable code.
+
+All it requires is a Temporal server to be [running locally](https://github.com/temporalio/temporal#download-and-start-temporal-server-locally):
+
+```sh
+brew install temporal
+temporal server start-dev
+```
+
+```python {title="temporal_agent.py" test="skip"}
+import uuid
+
+from temporalio import workflow
+from temporalio.client import Client
+from temporalio.worker import Worker
+
+from pydantic_ai import Agent
+from pydantic_ai.durable_exec.temporal import AgentPlugin, PydanticAIPlugin, TemporalAgent
+
+agent = Agent(
+    'gpt-5',
+    instructions="You're an expert in geography.",
+    name='geography',  # (10)!
+)
+
+temporal_agent = TemporalAgent(agent)  # (1)!
+
+
+@workflow.defn
+class GeographyWorkflow:  # (2)!
+    @workflow.run
+    async def run(self, prompt: str) -> str:
+        result = await temporal_agent.run(prompt)  # (3)!
+        return result.output
+
+
+async def main():
+    client = await Client.connect(  # (4)!
+        'localhost:7233',  # (5)!
+        plugins=[PydanticAIPlugin()],  # (6)!
+    )
+
+    async with Worker(  # (7)!
+        client,
+        task_queue='geography',
+        workflows=[GeographyWorkflow],
+        plugins=[AgentPlugin(temporal_agent)],  # (8)!
+    ):
+        output = await client.execute_workflow(  # (9)!
+            GeographyWorkflow.run,
+            args=['What is the capital of Mexico?'],
+            id=f'geography-{uuid.uuid4()}',
+            task_queue='geography',
+        )
+        print(output)
+        #> Mexico City (Ciudad de México, CDMX)
+```
+
+1. The original `Agent` cannot be used inside a deterministic Temporal workflow, but the `TemporalAgent` can.
+2. As explained above, the workflow represents a deterministic piece of code that can use non-deterministic activities for operations that require IO.
+3. [`TemporalAgent.run()`][pydantic_ai.durable_exec.temporal.TemporalAgent.run] works just like [`Agent.run()`][pydantic_ai.Agent.run], but it will automatically offload model requests, tool calls, and MCP server communication to Temporal activities.
+4. We connect to the Temporal server which keeps track of workflow and activity execution.
+5. This assumes the Temporal server is [running locally](https://github.com/temporalio/temporal#download-and-start-temporal-server-locally).
+6. The [`PydanticAIPlugin`][pydantic_ai.durable_exec.temporal.PydanticAIPlugin] tells Temporal to use Pydantic for serialization and deserialization, and to treat [`UserError`][pydantic_ai.exceptions.UserError] exceptions as non-retryable.
+7. We start the worker process that will listen on the specified task queue and run workflows and activities.
```

Comment:
Is this really a worker _process_? If it's not a distinct process, but just something else running on the same async event loop, I'd probably drop the word process here.

Separately, I would add here that in a production deployment, you would typically run the worker in a different code path than you'd use to execute a workflow. Basically just more clearly making the point that this structure is designed for making the example convenient to run and not what "real world" temporal code would look like.